### PR TITLE
Updated docker images to share scripts 

### LINKED ISSUE #42201: Update docker images to share scripts when possible
https://github.com/rust-lang/rust/pull/41958 added the possibility of sharing scripts used in docker images. This allows, for example, to use the same script to install sccache in all images, consequently  updating sccache can be done updating only one file.

We should update all images to share scripts when possible.

To update an image, one can follow these sequence of steps:

- Choose an unchecked image in the below list;
- Replace long `RUN` lines in the dockerfile of the image with scripts (like `sccache.sh` and `dum-init.sh` in `src/ci/docker/scripts/`, see for example the updates of [`arm-android`](https://github.com/rust-lang/rust/pull/41958/commits/307d8e5aa5f0c66881b1028fe1dc7f17e5ea317a#diff-c8faa107ca2811fa1a407474fce154bd) and [`dist-android`](https://github.com/rust-lang/rust/pull/41958/commits/307d8e5aa5f0c66881b1028fe1dc7f17e5ea317a#diff-6a6a986ec80374b59c335e7f3726e3f5)). You can organize the dockerfile in sections to make it easy to follow. sccache should be installed last. This avoids the whole container rebuild when sscache is updated;
- See if files in `COPY` lines are duplicated in other images (duplicated files can be found using the command `fdupes -r src/ci/`). If there are duplicated files, move then to `src/ci/docker/scripts/` and update the docker files;
- Test with the command `src/ci/docker/run.sh image` with the modified image still works (docker is required).

List of images (unchecked images need to be updated):

- [x] arm-android
- [ ] armhf-gnu
- [x] cross
- [ ] dist-aarch64-linux
- [x] dist-android
- [ ] dist-armhf-linux
- [ ] dist-arm-linux
- [ ] dist-armv7-linux
- [ ] dist-fuchsia
- [ ] dist-i586-gnu-i686-musl
- [ ] dist-i686-freebsd
- [ ] dist-i686-linux
- [ ] dist-mips64el-linux
- [ ] dist-mips64-linux
- [ ] dist-mipsel-linux
- [ ] dist-mips-linux
- [ ] dist-powerpc64le-linux
- [ ] dist-powerpc64-linux
- [ ] dist-powerpc-linux
- [ ] dist-s390x-linux
- [ ] dist-x86_64-freebsd
- [ ] dist-x86_64-linux
- [ ] dist-x86_64-musl
- [ ] dist-x86_64-netbsd
- [ ] emscripten
- [ ] i686-gnu
- [ ] i686-gnu-nopt
- [ ] x86_64-gnu
- [ ] x86_64-gnu-aux
- [ ] x86_64-gnu-debug
- [ ] x86_64-gnu-distcheck
- [ ] x86_64-gnu-full-bootstrap
- [ ] x86_64-gnu-incremental
- [ ] x86_64-gnu-llvm-3.7
- [ ] x86_64-gnu-nopt

### PR DESCRIPTION
**Pull request by Ophirr33**
Attempts to resolve #42201. I managed to pull out five scripts (android-base-apt-get, ubuntu16-apt-get, make3, rustbuild-setup, and crosstool-ng). Let me know if there's more I can do or if I should change some names.
r? @malbarbo

### CODE REVIEW

**Comment by Ophirr33** [comment]
Fixes #42201

**Comment by TimNN** [comment]
Is it possible that you forgot to commit the actual scripts? At least I can't find them in the Github diff.

**Comment by Ophirr33** [comment]
I did forget to push them, whoops. I just pushed them and then tried to squash the commits, but I don't think I'm squashing them correctly since it shows 3 commits that's trying to merge.

**Comment by TimNN** [comment]
@Ophirr33: The easiest thing is probably to just do a `git reset b695889` and then `git add` / `git commit` the changes again. Alternatively just let me know and I can do and push the rebase for you.

**Comment by Ophirr33** [comment]
Got it, thanks for the help @TimNN!

**Comment by arielb1** [comment]
What's the status of this? Are you waiting for @malbarbo to review this?

**Comment by Ophirr33** [comment]
@arielb1 Maybe? I just tagged @malbarbo because he created the issue. Haven't heard anything yet. 

_Sent from my Samsung SM-G935V using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by arielb1** [comment]
He does not even have r? rights.

 r? @alexcrichton

File: src/ci/docker/armhf-gnu/Dockerfile [hunk 1]
```diff
@@ -73,13 +73,14 @@ RUN arm-linux-gnueabihf-gcc addentropy.c -o rootfs/addentropy -static
 # TODO: What is this?!
 RUN curl -O http://ftp.nl.debian.org/debian/dists/jessie/main/installer-armhf/current/images/device-tree/vexpress-v2p-ca15-tc1.dtb
 
-RUN curl -o /usr/local/bin/sccache \
-      https://s3.amazonaws.com/rust-lang-ci/rust-ci-mirror/2017-05-12-sccache-x86_64-unknown-linux-musl && \
-      chmod +x /usr/local/bin/sccache
+# dumb-init
```
Comment (alexcrichton) [inline src/ci/docker/armhf-gnu/Dockerfile]: It's ok to remove these comments, I don't think they necessarily add much to what's already written down

File: src/ci/docker/scripts/ubuntu16-apt-get.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```
Comment (alexcrichton) [inline src/ci/docker/scripts/ubuntu16-apt-get.sh]: Mind renaming this to include the term "cross" in the script name? I think this list of packages is mostly suitable for building crosstool-ng and such and running that inside the container.

**Review by alexcrichton — COMMENTED** [review]
Thanks @Ophirr33!

**Comment by Ophirr33** [comment]
@alexcrichton Done!

**Pull request merged**